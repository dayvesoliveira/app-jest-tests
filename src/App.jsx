import React, { Fragment } from 'react'
import { connect } from 'react-redux'
import { BrowserRouter as Router } from 'react-router-dom'
// import { Crud } from './components/crud/list'

import Routes from './routes'
import { Provider } from 'react-redux'
import configureStore from './store'

const initStore = configureStore()

import { default as routePosts } from './routes/post.routes'


export class App extends React.Component {
    
    constructor(props){
        super(props)
    }

    render(){
        return (
            <Provider store={ initStore }>
                <Fragment>
                    <h1>UI<b>Faces</b></h1>
                    <label>
                        Username: 
                        <input placeholder="somenete para testes com jest" />
                    </label>
                    <br />
                    <br />
                    <Router>
                        <Routes routes={ routePosts }/>
                    </Router>
                </Fragment>
            </Provider>
        )
    }
}

export const mapStateToProps = state =>({ ...state })

export default connect(mapStateToProps)(App)