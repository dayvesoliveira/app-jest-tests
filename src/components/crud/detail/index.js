export { default as CrudDetail } from './container'
export { default as reducer } from './reducer'
 