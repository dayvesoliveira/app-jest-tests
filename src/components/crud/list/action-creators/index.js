import api from '../../../../config/api'

// definir aqui as actions

export const FETCH_SEARCH_DATA     = "FETCH_SEARCH_DATA"
export const FETCH_SEARCH_SUCCESS  = "FETCH_SEARCH_SUCCESS"
export const FETCH_SEARCH_FAILURE  = "FETCH_SEARCH_FAILURE"
export const RESET_SEARCH_DATA     = "RESET_SEARCH_DATA"

export const DELETE_REGISTER_DATA     = "DELETE_REGISTER_DATA"
export const DELETE_REGISTER_FAILURE  = "DELETE_REGISTER_FAILURE"

const END_POINT = '/posts'

export const loadSearchPosts = () => ({
   type: FETCH_SEARCH_DATA
})

export const setSearchPosts = list => ({
   type: FETCH_SEARCH_SUCCESS,
   list
})

export const errorResponsePosts = () => ({
   type: FETCH_SEARCH_FAILURE,
})

export const fetchSearchPosts = () => async dispatch =>{
    dispatch(loadSearchPosts())
    try {
        const response = await api.get(END_POINT)
        dispatch(setSearchPosts(response.data))
    } catch(e){
        dispatch(errorResponsePosts())
    }
}

export const resetPosts = () => ({
    type: RESET_SEARCH_DATA
})

export const errorDeletePost = () => ({
    type: DELETE_REGISTER_FAILURE
})

export const deletePost = id => async dispatch => {
	try {
		if (id) {
			dispatch(loadSearchPosts())

			await api.delete(`${END_POINT}/${id}`)

			dispatch(fetchSearchPosts())
		}
	} catch (er) {
		dispatch(errorDeletePost())
	}
}