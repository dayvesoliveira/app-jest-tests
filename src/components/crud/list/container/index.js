import React, { PureComponent, cloneElement } from 'react'
import PropTypes from 'prop-types'

import { 
    fetchSearchPosts, 
    resetPosts,
    deletePost,
    errorDeletePost
} from '../action-creators'

import GridPost from '../presentation'

import { connect } from 'react-redux'
import { selectCrud } from '../selector'

class Crud extends PureComponent {

    static propTypes = {
        fetchSearchPosts:   PropTypes.func.isRequired,
        searchPosts:        PropTypes.array
    }

    componentDidMount(){
        this.props.fetchSearchPosts()
    }

    handleDelete = async (id) =>  {
        try {
            if (!id) {
                throw Error('É obrigatório informa o id do registro.')
            } else {
                // implementar tela de confirmacao
                await this.props.deletePost(id)
            }
        } catch(e){}
    }

    render(){
        const { searchPosts } = this.props
        return (
            <GridPost 
                searchPosts={ searchPosts }
                handleDelete={ this.handleDelete }
                >
            </GridPost>
        )
    }
}

const mapStateToProps = state =>({
    ...selectCrud(state)
})

const actionCreators = {
	fetchSearchPosts,
    resetPosts,
    deletePost,
    errorDeletePost
}

export { Crud as CrudPureContainer }

export default connect(
	mapStateToProps,
	actionCreators
)(Crud)