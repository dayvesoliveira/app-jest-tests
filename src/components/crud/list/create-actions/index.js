import api from '../../../../config/api'

// definir aqui as actions

import api from '../../../../config/api'



export default { }