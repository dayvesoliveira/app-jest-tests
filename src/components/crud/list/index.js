export { default as Crud } from './container'
export { default as reducer } from './reducer' 