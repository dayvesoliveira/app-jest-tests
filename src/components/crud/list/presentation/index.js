import React from 'react'
import PropTypes from 'prop-types'
//import { Link } from 'react-router-dom'

import './styles.scss'
import ButtonLink from './button-link'

const GridPost = ({
    searchPosts,
    handleDelete
}) => (
        <table>
            <thead>
                <tr>
                    <th>#</th>
                    <th>Title</th>
                    <th>Body</th>
                    <th>User</th>
                    <th>Actions</th>
                </tr>
            </thead>
            <tbody>
                { searchPosts && searchPosts.map(post=>{
                    return (
                    <tr key={ 'pst_' + post.id }>
                        <td>{ post.id }</td>
                        <td>{ post.title }</td>
                        <td>{ post.body }</td>
                        <td>{ post.userId }</td>
                        <td>
                            <ButtonLink id={ post.id }/>
                            <button 
                                className="btn-exclude" 
                                onClick={()=>handleDelete(post.id)}>
                                Excluir
                            </button>
                        </td>
                    </tr>)
                })
                }
            </tbody>
        </table>
)

GridPost.propTypes = {
    searchPosts:  PropTypes.array,
    handleDelete: PropTypes.func.isRequired,
}

export default GridPost