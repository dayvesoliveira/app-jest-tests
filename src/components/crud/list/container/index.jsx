import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

import { fetchSearchPosts, resetPosts } from '../action-creators'
import GridPost from '../presentation'

import { connect } from 'react-redux'
import { selectCrud } from '../selector'

class Crud extends PureComponent {

    static propTypes = {
        fetchSearchPosts: PropTypes.func.isRequired,
        searchPosts: PropTypes.array,
        excludePost: PropTypes.func.isRequired
    }

    componentDidMount(){
        this.props.fetchSearchPosts()
    }

    handleDelete = id => {
        // codigo para executar o delete
        console.log(id)
    }

    render(){
        const { searchPosts } = this.props
        return (
            <GridPost 
                listPosts={ searchPosts }
                excludePost={ this.handleDelete }
                >
            </GridPost>
        )
    }
}

const mapStateToProps = state =>({
    ...selectCrud(state)
})

const actionCreators = {
	fetchSearchPosts,
    resetPosts
}

export { Crud as CrudPureContainer }

export default connect(
	mapStateToProps,
	actionCreators
)(Crud)