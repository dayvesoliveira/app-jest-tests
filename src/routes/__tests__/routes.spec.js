import React from 'react'
import { mount } from 'enzyme'
import { MemoryRouter } from 'react-router'
import Routes from '../index'
import { App } from '../../App'
import { Crud } from '../components/crud/list'

describe('routes using memory router', () => {

    it('should show Home component for / router (using memory router)', () => {
        const component = mount(
            <MemoryRouter initialEntries = {['/']} >
                <Routes />
            </MemoryRouter>
        )
        expect(component.find(App)).toHaveLength(1)
    })

    it('should show No match component for route not defined', () => {
        const component = mount(
            <MemoryRouter initialEntries = {['/posts','1']} >
                <Routes />
            </MemoryRouter>
        )
        expect(component.find(Crud)).toHaveLength(1)
    })

})