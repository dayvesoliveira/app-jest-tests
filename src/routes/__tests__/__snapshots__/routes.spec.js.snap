// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes using memory router should show Home component for / router (using memory router) 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      "/",
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "l8xpyj",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "l8xpyj",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Routes>
      <Suspense
        fallback={
          <div>
            Loading...
          </div>
        }
      >
        <Switch />
      </Suspense>
    </Routes>
  </Router>
</MemoryRouter>
`;
